### PR TITLE
Make the event list in the agenda view scrollable, calendar search

### LINKED ISSUE #6326: Agenda Items do not scroll in desktop mode
Repro:
- Have enough events in a day that there are more than fit onto the screen.
- You are not able to scroll to see the ones below.

Expected:
- Can scroll to see all events

We aren't using our own List component to display these, maybe use that here too?

## Test Notes
- [x] Event view popup is styled correctly
  - [x] Description icon & body are aligned
  - [x] Participants icon & body are aligned
  - [x] Invitation buttons look correct
  - [x] Stays correct at various view-port sizes
- [x] Event list in the agenda view scrolls
- [x] Details in the agenda view scrolls

### PR DESCRIPTION
**Pull request by rezbyte**
This refactors the agenda view to use three real columns instead of two columns styled to look like three. As a result, all three now scroll independently. Also cleaned up the relevant code via removing flow style variables names & refactoring.

Closes #6326.

### CODE REVIEW

**Review by murilopereirame — APPROVED** [review]

File: src/calendar/view/CalendarDesktopToolbar.ts [hunk 1]
```diff
@@ -14,36 +14,83 @@ type CalendarDesktopToolbarAttrs = {
 	onViewTypeSelected: (viewType: CalendarViewType) => unknown
 }
 
-export class CalendarDesktopToolbar implements Component<CalendarDesktopToolbarAttrs> {
-	view({ attrs }: Vnode<CalendarDesktopToolbarAttrs>): Children {
-		const { navConfig } = attrs
+type CalendarDesktopHeadingToolbarAttrs = {
+	navConfig: CalendarNavConfiguration
+	isWithWrapper?: boolean
```
Comment (charlag) [inline src/calendar/view/CalendarDesktopToolbar.ts:19]: what is `isWithWrapper` and why is it always `false`

Comment (charlag) [reply src/calendar/view/CalendarDesktopToolbar.ts:19]: oh I see why it looks like it is, it is negated

File: src/calendar/view/CalendarDesktopToolbar.ts [hunk 2]
```diff
@@ -86,17 +137,20 @@ export class CalendarDesktopToolbar implements Component<CalendarDesktopToolbarA
...
 		)
 	}
+}
 
-	private renderWeekNumberLabel(label: string): Children {
+export class CalendarDesktopToolbar implements Component<CalendarDesktopToolbarAttrs> {
+	view({ attrs }: Vnode<CalendarDesktopToolbarAttrs>): Children {
 		return m(
-			".ml-m.small.font-weight-600",
+			".flex.row.items-center.content-bg.border-radius-big.mlr-l.rel.pr.pl-vpad.button-height",
```
Comment (charlag) [inline src/calendar/view/CalendarDesktopToolbar.ts:143]: toolbar used  to be height "50", now it's `button-height`

File: src/calendar/view/CalendarView.ts [hunk 3]
```diff
@@ -757,6 +816,19 @@ export class CalendarView extends BaseTopLevelView implements TopLevelView<Calen
...
+			const switchingFromAgenda = oldViewType === CalendarViewType.AGENDA && this.currentViewType !== CalendarViewType.AGENDA
+			const switchingToAgenda = oldViewType !== CalendarViewType.AGENDA && this.currentViewType === CalendarViewType.AGENDA
+			if (switchingFromAgenda || switchingToAgenda) {
+				console.log("Changing columns!")
+				this.viewSlider = new ViewSlider(
+					this.currentViewType === CalendarViewType.AGENDA
+						? [this.sidebarColumn, this.listColumn, this.detailsColumn]
+						: [this.sidebarColumn, this.listColumn],
+				)
+				m.redraw()
```
Comment (charlag) [inline src/calendar/view/CalendarView.ts:775]: wouldn't `onNewUrl` always cause a redraw anyway?

File: src/gui/main-styles.ts [hunk 4]
```diff
@@ -332,6 +332,9 @@ styles.registerStyle("main", () => {
 		".ml-l": {
 			"margin-left": px(size.hpad_large),
 		},
+		".ml-a": {
```
Comment (charlag) [inline src/gui/main-styles.ts:335]: maybe `.ml-auto`

File: src/gui/nav/ViewSlider.ts [hunk 5]
```diff
@@ -29,6 +29,7 @@ export const gestureInfoFromTouch = (touch: Touch): GestureInfo => ({
 interface ViewSliderAttrs {
 	header: Children
 	bottomNav: Children
+	isAllOrMain?: boolean
```
Comment (charlag) [inline src/gui/nav/ViewSlider.ts:32]: I'm not sure I understand what it is or what it's for

**Comment by ganthern** [comment]
I merged the calendar search in here to prevent mistakes when we actually go to integrating everything

**Pull request merged**